src/test/java/uk/ac/cam/cl/dtg/segue/api/AdminFacadeTest.java

```diff
@@ -0,0 +1,423 @@
+        }
+
+        verify(userManager);
+      }
+
+      @Test
+      void modifyUsersTeacherPendingStatus_unknownTarget_returnsBadRequest()
+          throws NoUserLoggedInException, SegueDatabaseException, NoUserException {
+        HttpServletRequest mockRequest = replayMockServletRequest();
+        RegisteredUserDTO currentUser = new RegisteredUserDTO();
+        List<Long> targetUsers = List.of(2L);
+
+        expect(userManager.getCurrentRegisteredUser(mockRequest)).andReturn(currentUser);
+        expect(isUserAnAdminOrEventManager(userManager, currentUser)).andReturn(true);
+
+        expect(userManager.getUserDTOById(2L)).andThrow(new NoUserException("No user found with this ID!"));
+
+        replay(userManager);
+
+        try (Response response = adminFacade.modifyUsersTeacherPendingStatus(mockRequest, false, targetUsers)) {
+          assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+          assertEquals("One or more users could not be found: [2]. ",
+              response.readEntity(SegueErrorResponse.class).getErrorMessage());
+        }
+
+        verify(userManager);
+      }
+
+      @Test
+      void modifyUsersTeacherPendingStatus_failureForOneOfMultipleTargets_returnsBadRequest()
+          throws NoUserLoggedInException, SegueDatabaseException, NoUserException, ContentManagerException {
+        HttpServletRequest mockRequest = replayMockServletRequest();
+        RegisteredUserDTO currentUser = new RegisteredUserDTO();
+        List<Long> targetUsers = List.of(2L, 3L);
+
+        expect(userManager.getCurrentRegisteredUser(mockRequest)).andReturn(currentUser);
+        expect(isUserAnAdminOrEventManager(userManager, currentUser)).andReturn(true);
+
+        expect(userManager.getUserDTOById(2L)).andThrow(new NoUserException("No user found with this ID!"));
+
+        expect(userManager.getUserDTOById(3L)).andReturn(prepareTestUser(3L, true));
+        expect(userManager.updateTeacherPendingFlag(3L, false)).andReturn(prepareTestUser(3L, false));
+
+        expect(emailManager.getEmailTemplateDTO("teacher_declined")).andReturn(new EmailTemplateDTO());
+        emailManager.sendTemplatedEmailToUser(anyObject(RegisteredUserDTO.class), anyObject(EmailTemplateDTO.class),
+            anyObject(Map.class), eq(EmailType.SYSTEM));
+        expectLastCall();
+
+        replay(userManager);
+        replay(emailManager);
+
+        try (Response response = adminFacade.modifyUsersTeacherPendingStatus(mockRequest, false, targetUsers)) {
+          assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+          assertEquals("One or more users could not be found: [2]. ",
+              response.readEntity(SegueErrorResponse.class).getErrorMessage());
+        }
+
+        verify(userManager);
+        verify(emailManager);
+      }
+
+      @Test
+      void modifyUsersTeacherPendingStatus_emailSendFailureForOneTarget_returnsBadRequest()
```

Comment:
As mentioned in my other comment, it sounds like this should be a server error instead.

Reply:
On second thought, how important is this, actually? Maybe this is acceptable? This is more of a business logic question: if we fail to send some e-mails, what does that mean for the end user? Should we be returning a 200 response, but with a warning message or something?

Reply:
If 200 response comes back that's "ok" so no message comes back to the end user on the front end (and they aren't inspecting the network response). 

If it returns an error code under 500 it doesn't totally bomb out the page, it shows an error popup with extra information. So I was trying to do this in a user friendly way so that a user might get a popup that notifies of email send failures (for whatever reason) but the teacher_pending status of those users in the DB would still have been modified. In which case the user (who is a STEM staff member) can manually send an email. 

If we returned a 500 error it would takeover the page, and be unclear to the user that it had updated the statuses and just failed at the email send point.

Reply:
Thanks for explaining, Hannah. That makes sense, but you're seeing all this from the perspective of the current implementation of the frontend, however when designing the API you need to make sure that it adheres to the conventions that RESTful APIs normally follow. For instance, the server failing to send some emails is not the client's fault, so you cannot return a 4xx error. It should be a 500 error, since it's a server error. However, if sending these emails are not completely necessary, one could argue that if the server manages to update all users' teacher pending status, but it failed to send some emails, then that could be considered a 200 response, because the main task (updating the status) was successful. In the body of that response, you could have a JSON with an optional attribute called `warning`, or something like that, and populate that with the IDs of the users that you didn't manage to send the emails to, for instance. Does that make sense?

Reply:
Changed this now so that email send failure sends back 200 response with a message (and front end adjusted to display that message in a popup).